code/Path_Distance_TableGenerator.py

```diff
@@ -0,0 +1,311 @@
+from shortest_physical_path import parse_arguments
+import sqlite3
+import networkx as nx
+from networkx.exception import NetworkXNoPath
+from collections import defaultdict
+import Querying_Database as qdb
+from shapely.geometry import Point
+from shapely import wkt
+import ast
+import sys
+    
+global_graph = nx.Graph()
+global_edges_dict =  {}
+global_nodes_dict = {}
+global_querier = None
+
+
+def get_distance_between_two_router(from_place, to_place):
+    src = from_place
+    dst = to_place
+    route = None
+    dist = 0.0
+    route_geom = []
+    waypoints_geom = []
+
+    cities_valid = are_cities_valid(src, dst) 
+    if cities_valid:
+        route = get_shortest_path(src, dst, route_geom, waypoints_geom)
+        if route:
+            dist = calc_dist_along_path(route)
+        else:
+            dist = 0
+    return dist
+
+def are_cities_valid(src, dst):
+    if len(src.split(',')) == 3:
+        fc = src.split(',')[0].strip()
+        fs = src.split(',')[1].strip()
+        fcc = src.split(',')[2].strip()
+        f_query = f"""SELECT * 
```

Comment:
It looks like we're calling this function a lot, for every city in a shortest path. Since we've loaded this city_points table into nx graph, can we use nx graph to look up the node? In memory should be faster. But if this is not taking too long, then maybe it's fine not to change that.

Reply:
It is super fast for this part: less than 1 second I guess, the time overhead is mainly for building up the graph